src/createRoom.ts

```diff
@@ -398,16 +398,21 @@ export default async function createRoom(opts: IOpts): Promise<string | null> {
 export async function canEncryptToAllUsers(client: MatrixClient, userIds: string[]): Promise<boolean> {
     try {
         const usersDeviceMap = await client.downloadKeys(userIds);
-        // { "@user:host": { "DEVICE": {...}, ... }, ... }
-        return Object.values(usersDeviceMap).every(
-            (userDevices) =>
-                // { "DEVICE": {...}, ... }
-                Object.keys(userDevices).length > 0,
-        );
+
+        // There are no devices at all.
```

Comment:
s/devices/users/.

I think this is a change of behaviour from before 3.69.0. Are we sure it's what's expected?

It's also inconsistent with the documentation on the function. If there are *no* users in the room (or rather, in the `userIds` list) then "there is at least one device that we can encrypt to" is trivially true.

Reply:
> It's also inconsistent with the documentation on the function. If there are _no_ users in the room (or rather, in the `userIds` list) then "there is at least one device that we can encrypt to" is trivially true.

sorry, I misquoted.

If there are *no* users, then "for every user in a room, \<X>" is trivially true, for any \<X>.

Reply:
Yes, this is a change. Not sure if it is correct. If I download the keys for a list of users and I do get an empty response, can I not encrypt to them? :thinking:

An empty `userIds` arg is a slightly separate case. If I do not specify a user, I cannot encrypt to them.

Reply:
If a user has no devices, I don't think I want that to make me stop sending to a whole list of users. I can trivially do the right thing for that user, which is nothing.

Happy to be persuaded though.

Reply:
@andybalaam do you mean a misc case here where some users have devices?

The discussion from above is about this case:
- Before: 0 users or 0 devices returned by `downloadKeys` → `true`
- Now: `false`

Reply:
https://github.com/vector-im/element-web/issues/24998

Reply:
> If a user has no devices, I don't think I want that to make me stop sending to a whole list of users. I can trivially do the right thing for that user, which is nothing.

First of all, let's clarify. This code is relying on `downloadKeys` only returning devices which have encryption keys. So we're talking about *encryption-capable* devices here.

The whole point of this function is to check that all users in a given list have at least one encryption-capable device - which means we can safely encrypt a room which contains those users.

So, I disagree with you here @andybalaam: if any one of the users in the list has no (encryption-capable) devices, I do *not* want to start an encrypted room with the whole list.

Reply:
What Rich said makes perfect sense to me.